### PR TITLE
Documentation: "Tutorials section >> docs"

### PR DESCRIPTION
**Pull request by daveads**
* Did some Restructuring the sidebar and made sidebar.js a bit easier to read
* created a Tutorial section and move **"Testing HTTP Applications"** into it 
* Added **DocCardList** to Reference and Example section 

DocCardList looks like the below 
Screenshot 2024-04-25 at 16-38-07 Zio http Reference zio-http-docs-website

### CODE REVIEW

**Comment by 987Nabil** [comment]
@khajavi can you take a look please

**Comment by khajavi** [comment]
@daveads Just ping me to review after resolving the conflicts.

**Comment by daveads** [comment]
@khajavi you can check it out now...

**Comment by daveads** [comment]
we will move overveiw which was formally index.md later just like you earlier discussed 

Screenshot 2024-04-27 at 14-01-47 Zio http Reference zio-http-docs-website

File: docs/examples/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+id: index
+title: "Zio http Examples"
```
Comment (khajavi) [inline docs/examples/index.md]:
```suggestion
title: "ZIO HTTP Examples"
sidebar_label: Examples
```

File: docs/reference/index.md [hunk 2]
```diff
@@ -1,246 +1,10 @@
 ---
 id: index
-title: Introduction to ZIO HTTP
+title: "Zio http Reference"
```
Comment (khajavi) [inline docs/reference/index.md]:
```suggestion
title: "ZIO HTTP Reference"
sidebar_label: Reference
```

File: docs/sidebars.js [hunk 3]
```diff
@@ -4,102 +4,133 @@ const sidebars = {
...
           collapsed: true,
           link: { type: "doc", id: "reference/index" },
           label: "Reference",
           items: [
-            "reference/server",
-            "reference/client",
+            "reference/overview",
+            "reference/server", 
+            "reference/client", 
+            "reference/handler",
```
Comment (khajavi) [inline docs/sidebars.js]: I would suggest placing the handler after the routing section for better organization.

**Comment by khajavi** [comment]
@daveads Added some suggestions and comments.

**Review by khajavi — APPROVED** [review]

**Comment by daveads** [comment]
> @daveads Added some suggestions and comments.

Done

**Review by 987Nabil — APPROVED** [review]

**Pull request merged**